Strats/Fibonacci/FibonacciMultiStock.py

```diff
@@ -0,0 +1,145 @@
+        #print("calculating")
+        low = (10**9, 0)
+        high = (-10**9, 0)
+        for i in range(self.scope):
+            #print(self.data[ticker][(i+1)*-1], i)
+            if self.data[ticker][(i+1)*-1] > high[0]:
+                #print("found a new high at i", i)
+                high = (self.data[ticker][(i+1)*-1], i)
+            if self.data[ticker][(i+1)*-1] < low[0]:
+                #print("found a new low at i", i)
+                low = (self.data[ticker][(i+1)*-1], i)
+        #print("low is ", low[0], "at i ", low[1])
+        #print("high is", high[0], "at i ", high[1])
+        #incline
+        if low[1] > high[1]:
+            #print("here hehe xd")
+            l, r = low[0], high[0]
+            self.bands[ticker][0] = r
+            self.bands[ticker][-1] = l
+            for i in range(len(self.ratios)):
+                self.bands[ticker][len(self.bands)-1-i] = l + (r-l) * self.ratios[i]
+
+        #decline
+        elif low[1] < high[1]:
+            #print("here at", self.ticks)
+            l, r = high[0], low[0]
+            self.bands[ticker][0] = r
+            self.bands[ticker][-1] = l
+            for i in range(len(self.ratios)):
+                self.bands[ticker][len(self.bands)-1-i] = l + (r-l) * self.ratios[i]
+
+    def updateSingle(self, **kwargs):
+        """
+        Update method called every tick. Just takes in the stock's price at the time of the tick
+        """
+        #updating for the nth stock
+        #if the tick we're at is equal to our defined time for when the next "retracement" period starts, make note
+        #of it and create our new bands
+        ticker = kwargs["ticker"]
+        price = kwargs["price"]
+
+        if self.activeRetracement:
+            prev = self.data[ticker][-1]
+            #print(prev, price)
+            #incline
+            if self.bands[ticker][0] > self.bands[ticker][-1]:
+                #check if our current price compared w our prev price has crossed any of the bounds
+                for band in self.bands[ticker]:
+                    if prev >= band and price <=band:
+                        for i in range(self.ordersWhenHitsBand):
+                            #if before we were inclined, then we started going down and bounced off a band,
+                            #add a buy order
+                            self.orders[ticker].append("BUY")
+            #decline
+            elif self.bands[ticker][0] < self.bands[ticker][-1]:
+                for band in self.bands[ticker]:
+                    if prev <= band and price >= band:
+                        for i in range(self.ordersWhenHitsBand):
+                            #if before we declined, then started going up and bounced off a band, add a sell order                      
+                            self.orders[ticker].append("SELL")
+        self.data[ticker].append(price)
+        return self.orders
+        
+
+    def updateAll(self, newData):
```

Comment:
please describe new data with a docstring and rename this function to update

Reply:
Done